lib/parse.js

```diff
@@ -83,6 +84,9 @@ var parseValues = function parseQueryStringValues(str, options) {
         var key, val;
         if (pos === -1) {
             key = options.decoder(part, defaults.decoder, charset, 'key');
+            if (key === '') {
+                continue;
```

Comment:
handles lonely `&`

Reply:
what if the key already exists? this would bypass the `utils.combine` logic below

Reply:
Could you provide input for this case? All previous inputs are still valid and i've covered this cases in tests:
```
&
&=
&=&
&=&=
&=&=&
=
=&
=&&&
=&=&=&
``` 

Do i miss something? Thanks

Reply:
I don't have any input off the top of my head :-) just looking at the code paths.